terraform/sqs.tf.json.template.py

```diff
@@ -18,10 +18,13 @@
                             "visibility_timeout_seconds": config.contribution_lambda_timeout(retry=retry) + 10,
                             "message_retention_seconds": 24 * 60 * 60,
                             "redrive_policy": json.dumps({
-                                "maxReceiveCount": 9 if retry else 1,
-                                "deadLetterTargetArn": "${aws_sqs_queue.%s.arn}"
-                                                       % config.unqual_notifications_queue_name(retry=not retry,
-                                                                                                fail=retry)
+                                "maxReceiveCount": 1,
+                                "deadLetterTargetArn": "${aws_sqs_queue.%s.arn}" % (
+                                    config.unqual_redrive_queue_name()
+                                    if retry else
+                                    config.unqual_notifications_queue_name(retry=not retry,
+                                                                           fail=retry)
```

Comment:
I think this becomes a lot simpler and easier to read if we avoid the variable, whose value is known at this point anyway due to the if/else:
```suggestion
                                    config.unqual_notifications_queue_name(retry=True,
                                                                           fail=False)
```

Reply:
Absolutely agree, however the variable is iterated over in a loop. Your suggestion sets the fail-notifications queue as the deadletter queue for the non retry notifications queue, which isn't the desired behavior. We're sacrificing readability for reducing duplication.

Reply:
I'm not convinced that your explanation is correct. The value of `retry` does change throughout the loop, but the expression on line 25 is only evaluated when `retry` is `False` due to the ternary if/else on line 24. I compared the emitted TF configuration and confirmed that my approach generates identical output:
```
$ # Starting with HEAD
$ git diff
$ # Generate TF config from templates
$ make -C terraform/ plan
...
$ # Create a backup of the original config
$ cp terrafform/sqs.json copy.json
$ # Edit file
$ git diff
diff --git a/terraform/sqs.tf.json.template.py b/terraform/sqs.tf.json.template.py
index 791201d5..20422920 100644
--- a/terraform/sqs.tf.json.template.py
+++ b/terraform/sqs.tf.json.template.py
@@ -22,7 +22,7 @@ emit_tf(
                                 "deadLetterTargetArn": "${aws_sqs_queue.%s.arn}" % (
                                     config.unqual_redrive_queue_name()
                                     if retry else
-                                    config.unqual_notifications_queue_name(retry=not retry, fail=retry)
+                                    config.unqual_notifications_queue_name(retry=True, fail=False)
                                 )
                             })
                         }
$ # Regenerate TF config with the code changes shown from the previous command
$ make -C terraform/ plan
$ # Confirm the generated content has not changed
$ diff -u copy.json terraform/sqs.json  
$
```
If you have continued doubts or concerns about this, let's defer to Hannes.

Reply:
Yeah, I made a mistake that lead me to the wrong conclusion, I appreciate the thorough review.